pkg/providers/gcp/gcp.go

```diff
@@ -152,33 +164,35 @@ func (c *Client) Refresh(ctx context.Context, project string) {
 				continue
 			}
 
+			kind, err := getValueFromURL(instance.GetMachineType())
+			if err != nil {
+				logger.Error("failed to get instance type from url")
+			}
+
+			mapInstance := &v1.Instance{
+				ID:       instanceID,
+				Provider: provider,
+				Name:     name,
+				Region:   region,
+				Zone:     zone,
+				Service:  service,
+				Kind:     kind,
+				Labels: v1.Labels{
+					"Lifecycle": instance.GetScheduling().GetProvisioningModel(),
+					"ID":        instanceID,
+				},
+			}
 			if instance.GetStatus() == "TERMINATED" {
-				// delete the entry from the cache
-				delete(c.instancesMap, util.Key(zone, service, name))
+				mapInstance.Status = v1.Terminated
```

Comment:
I think instead of iterating all instances, we can just check here and remove if it's status was already set to terminated (then it means we previously looped through metrics and pushed to the bus) 

so like 
```
if instance.GetStatus() == "TERMINATED" {
    if mapInstance.Status == v1.Terminated {
          DELETE
          continue
    }
    mapInstance.Status = v1.Terminated